src/datasets/table.py

```diff
@@ -0,0 +1,591 @@
+    def column(self, *args, **kwargs):
+        return self.table.column(*args, **kwargs)
+
+    @wraps_arrow_table_method(pa.Table.itercolumns)
+    def itercolumns(self, *args, **kwargs):
+        return self.table.itercolumns(*args, **kwargs)
+
+    @property
+    def schema(self):
+        return self.table.schema
+
+    @property
+    def columns(self):
+        return self.table.columns
+
+    @property
+    def num_columns(self):
+        return self.table.num_columns
+
+    @property
+    def num_rows(self):
+        return self.table.num_rows
+
+    @property
+    def shape(self):
+        return self.table.shape
+
+    @property
+    def nbytes(self):
+        return self.table.nbytes
+
+    @property
+    def column_names(self):
+        return self.table.column_names
+
+    def __getitem__(self, i):
+        return self.table[i]
+
+    def __len__(self):
+        return len(self.table)
+
+    def __repr__(self):
+        return self.table.__repr__().replace("pyarrow.Table", self.__class__.__name__)
+
+    def __str__(self):
+        return self.table.__str__().replace("pyarrow.Table", self.__class__.__name__)
+
+    @wraps_arrow_table_method(pa.Table.slice)
+    def slice(self, *args, **kwargs):
+        raise NotImplementedError()
+
+    @wraps_arrow_table_method(pa.Table.filter)
+    def filter(self, *args, **kwargs):
+        raise NotImplementedError()
+
+    @wraps_arrow_table_method(pa.Table.flatten)
+    def flatten(self, *args, **kwargs):
+        raise NotImplementedError()
+
+    @wraps_arrow_table_method(pa.Table.combine_chunks)
+    def combine_chunks(self, *args, **kwargs):
+        raise NotImplementedError()
+
+    @wraps_arrow_table_method(pa.Table.cast)
+    def cast(self, *args, **kwargs):
+        raise NotImplementedError()
+
+    @wraps_arrow_table_method(pa.Table.add_column)
+    def add_column(self, *args, **kwargs):
+        raise NotImplementedError()
+
+    @wraps_arrow_table_method(pa.Table.append_column)
+    def append_column(self, *args, **kwargs):
+        raise NotImplementedError()
+
+    @wraps_arrow_table_method(pa.Table.remove_column)
+    def remove_column(self, *args, **kwargs):
+        raise NotImplementedError()
+
+    @wraps_arrow_table_method(pa.Table.set_column)
+    def set_column(self, *args, **kwargs):
+        raise NotImplementedError()
+
+    @wraps_arrow_table_method(pa.Table.rename_columns)
+    def rename_columns(self, *args, **kwargs):
+        raise NotImplementedError()
+
+    @wraps_arrow_table_method(pa.Table.drop)
+    def drop(self, *args, **kwargs):
+        raise NotImplementedError()
+
+
+class TableBlock(Table):
+    """
+    TableBlock is the allowed class inside a ConcanetationTable.
+    Only MemoryMappedTable and InMemoryTable are TableBlock.
+    This is because we don't want a ConcanetationTable made out of other ConcanetationTables.
+    """
+
+    pass
+
+
+class InMemoryTable(TableBlock):
+    """
+    Wraps a pyarrow Table using composition.
```

Comment:
I'm not sure this line is very helpful, for `MemoryMappedTable` as well.

Reply:
will add more context in the docstring, especially regarding memory handling with pyarrow